bin/reth/src/commands/import_receipts.rs

```diff
@@ -97,47 +92,83 @@ impl ImportReceiptsCommand {
             );
         }
 
+        // prepare the tx for `write_to_storage`
         let tx = provider.into_tx();
+        let mut total_decoded_receipts = 0;
+
+        // open file
+        let mut reader = ChunkedFileReader::new(&self.path, self.chunk_len).await?;
 
         while let Some(file_client) = reader.next_chunk::<ReceiptFileClient>().await? {
             // create a new file client from chunk read from file
+            let ReceiptFileClient {
+                mut receipts,
+                mut first_block,
+                total_receipts: total_receipts_chunk,
+            } = file_client;
+
+            // mark these as decoded
+            total_decoded_receipts += total_receipts_chunk;
+
             info!(target: "reth::cli",
+                first_receipts_block=?first_block,
+                total_receipts_chunk,
                 "Importing receipt file chunk"
             );
 
-            let ReceiptFileClient { receipts, first_block, total_receipts: total_receipts_chunk } =
-                file_client;
-
-            total_decoded_receipts += total_receipts_chunk;
-
+            // It is possible for the first receipt returned by the file client to be block #1
+            // rather than the genesis block. In this case, we just prepend empty receipts to the
+            // current list of receipts. When initially writing to static files, the provider
+            // expects the first block to be block zero. So, if the first block returned by the file
+            // client is block #1, we insert empty receipts for the genesis block.
```

Comment:
is this just for the first block, or is it true in general that blocks and receipt lists are expected to map 1-1 for code that handles insertion to static files?

Reply:
> is this just for the first block, or is it true in general that blocks and receipt lists are expected to map 1-1 for code that handles insertion to static files?

Sort of both, if there aren't any receipts (ie if its `vec![]`) nothing gets added but the list still has to exist so `increment_block` is called. this check is related to using `first_block`, if we start with block 1 and don't have anything in static files, this will be called and we'll panic:
https://github.com/paradigmxyz/reth/blob/1c73777bf8cbd5c304e248c2ae2d83607395d72c/crates/storage/provider/src/providers/static_file/writer.rs#L242-L248